lib/game_logic.rb

```diff
@@ -0,0 +1,209 @@
+require_relative '../bin/main'
+
+include UserInteraction
+#====================================================================
+# This is the class Player and is called every time a new game starts
+#====================================================================
+class Player
+    public
+    def validate_rol
+        get_rol
+        if @rol === 1 || @rol === 2
+            board = Board.new ["1","2","3","4","5","6","7","8","9"], @rol, @name_1, @name_2
+            board.print_board
+            
+        else
+            puts "\n\n\nNumber #{@rol} is not a valid option"
+            validate_rol
+        end
+    end
+end
+
+
+
+#===============================================================================================================================================
+# This is the class Rules. This class contains all rules of the game, so everytime a player plays, this class will verify if movements are valid
+# This class also checks if there is a winner or in case the game ends it will puts Game Over
+#===============================================================================================================================================
+
+class Rules < Player
+    def winner?
+        #Horizontal Test
+        
+        if @board[0] === @draw && @board[1] === @draw && @board[2] === @draw
+            @board[0] = "-" 
+            @board[1] = "-" 
+            @board[2] = "-"
+            print_last
+            puts "\nPlayer #{@name} Wins"
+            again?
+        elsif @board[3] === @draw && @board[4] === @draw && @board[5] === @draw
+            @board[3] = "-" 
+            @board[4] = "-" 
+            @board[5] = "-"
+            print_last
```

Comment:
`print_last` is not defined in that class.
It's defined in `Board` class. You can't call it like this.

You must separated the class. I suggest you put them in their own file to see how you implementation is not working.